### DISCUSSION TITLE
Declarative routes and api doc generation

### DISCUSSION THREAD

**Discussion by HuiiBuh** [fastapi/fastapi#7596] [Questions] [answered]
### First check

* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

I have an application which uses the declarative routing used by starlette. 
The setup is the following.

URLs.py
```python
from fastapi.routing import APIRoute

playlist_routes = [
    APIRoute('/{playlist_id}/tracks', get_playlist_tracks, methods=['GET']),
    APIRoute('/{playlist_id}', get_one_playlist, methods=['GET']),
]
# ...
```

main.py
```python
from .URLs import playlist_routes

routes = [
    Mount('/playlists/', routes=playlist_routes)
]
# ...
app = FastAPI(routes=routes)
```
Now I wanted to visit the generated api docs, but fastapi does not pick up the created routes. 

![image](https://user-images.githubusercontent.com/46561026/81508572-239f8d80-9305-11ea-8edd-e05a06b5e96c.png)

Is it somehow possible to get the api doc generation to work with this declarative routing style?

**Comment by tiangolo** [comment, marked as answer]
You are using a mount, that creates a whole independent app, you have to use `include_router` to include the path operations from another router in the same app.

It would be better to use the default/standard style with decorators, that way you can just follow the docs to find the answers.

**Comment by HuiiBuh** [comment]
Thank you for the answer.
